public/app/features/explore/spec/helper/interactions.ts

```diff
@@ -33,6 +33,28 @@ export const openQueryHistory = async (exploreId: ExploreId = ExploreId.left) =>
   ).toBeInTheDocument();
 };
 
+export const closeQueryHistory = async (exploreId: ExploreId = ExploreId.left) => {
+  const closeButton = withinExplore(exploreId).getByRole('button', { name: 'Close query history' });
+  userEvent.click(closeButton);
+};
+
+export const switchToQueryHistoryTab = async (
+  name: 'Settings' | 'Query History',
+  exploreId: ExploreId = ExploreId.left
+) => {
+  userEvent.click(withinExplore(exploreId).getByLabelText(`Tab ${name}`));
+};
+
+export const selectStarredTabFirst = (exploreId: ExploreId = ExploreId.left) => {
+  const checkbox = withinExplore(exploreId).getByLabelText(/Change the default active tab.*/);
+  userEvent.click(checkbox);
+};
+
+export const selectOnlyActiveDataSource = (exploreId: ExploreId = ExploreId.left) => {
+  const checkbox = withinExplore(exploreId).getByLabelText(/Only show queries for data source currently active.*/);
+  userEvent.click(checkbox);
+};
+
```

Comment:
we can query by the checkbox role+its accessible name rather than just by its label:
```ts
const checkbox = withinExplore(exploreId).getByRole('checkbox', {
    name: 'Change the default active tab from “Query history” to “Starred”',
  });
```

also, why the wildcard match?

Reply:
> also, why the wildcard match?

I thought it may be more readable when it's shorter but it's actually confusing. Changed in 5be9baa21e7b517b2e29f8af9dbec4320db405d8